### PR TITLE
React to options

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public RoutingBenchmark()
                 NullLoggerFactory.Instance,
                 UrlEncoder.Default,
                 new DefaultObjectPool<UriBuildingContext>(new UriBuilderContextPooledObjectPolicy(UrlEncoder.Default)),
-                new DefaultInlineConstraintResolver(new OptionsManager<RouteOptions>(Enumerable.Empty<IConfigureOptions<RouteOptions>>())));
+                new DefaultInlineConstraintResolver(new OptionsManager<RouteOptions>(new OptionsFactory<RouteOptions>(Enumerable.Empty<IConfigureOptions<RouteOptions>>(), Enumerable.Empty<IPostConfigureOptions<RouteOptions>>()))));
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: Would it be possible to add a default constructor to this? We have a million places in MVC where we've done stuff like this

Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: Like **literally** a million

Comment (HaoK) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: Are these all test cases that are bypassing DI?

Comment (rynowak) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: There are literally a million places that MVC constructs an options manager

Comment (HaoK) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: Well taking a look at MVC now

Comment (HaoK) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: It doesn't look THAT bad, WebAPIShim is the only one not using TestOptionsMonitor, which is how things are supposed to be used.

**Pull request merged**